app/src/main/java/com/babylon/wallet/android/domain/usecases/NotariseTransactionUseCase.kt

```diff
@@ -47,7 +48,11 @@ class NotariseTransactionUseCase @Inject constructor() {
         }
 
         val signatures = signatureGatherer.gatherSignatures(intent = intent).getOrElse { error ->
-            return Result.failure(PrepareTransactionException.SignCompiledTransactionIntent(error))
+            if (error is RadixWalletException.DappRequestException.RejectedByUser) {
```

Comment:
this is really getting confusing, if we do "internal" transfer and fail gathering signatures because of cancelling, this will be thrown? We should go over the exception structure and try to make it more maintanable.

Reply:
indeed we must tackle the error/exception handling. We need a ticket and a plan for it.

I think I tested the internal transfer already, but I will give it one more try.